lib/src/rules/enable_null_safety.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Do use sound null safety.';
+
+const _details = r'''
+**DO** use sound null safety, by not specifying a dart version lower than `2.12`.
+
+**BAD:**
+```dart
+// @dart=2.8
+a() {
+}
+```
+
+**GOOD:**
+```dart
+b() {
+}
+''';
+
+class EnableNullSafety extends LintRule implements NodeLintRule {
+  EnableNullSafety()
+      : super(
+            name: 'enable_null_safety',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this, context);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+  final LinterContext context;
+  // to be kept in sync with LanguageVersionOverrideVerifier (added groups for the version)
+  final regExp = RegExp(r'^\s*//\s*@dart\s*=\s*(\d+)\.(\d+)');
+
+  _Visitor(this.rule, this.context);
+
+  @override
+  void visitCompilationUnit(CompilationUnit node) {
+    CommentToken? comment = node.beginToken.precedingComments;
```

Comment:
I didn't find any documentation that explicitly mentions it, but I think it's valid to have both a script tag and a language version override comment in the same file, and I'm guessing that the comment follows the script tag.

That's an unlikely corner case, but if I'm right then the `beginToken` might be a script tag token, in which case you would need to skip to the next token before getting the `precedingComments`.